ethdb/redisdb/batch.go

```diff
@@ -0,0 +1,83 @@
+package redisdb
+
+import (
+	"github.com/ethereum/go-ethereum/ethdb"
+)
+
+// batch is a write-only memory batch that commits changes to its host
+// database when Write is called. A batch cannot be used concurrently.
+type batch struct {
+	db      *Database
+	deletes []string
+	writes  []string
+	size    int
+}
+
+// Put inserts the given value into the batch for later committing.
+func (b *batch) Put(key, value []byte) error {
+	b.writes = append(b.writes, string(key), string(value))
+	b.size += len(key) + len(value)
+	return nil
+}
+
+// Delete inserts the a key removal into the batch for later committing.
+func (b *batch) Delete(key []byte) error {
+	if len(key) == 0 {
+		return nil
+	}
+	b.deletes = append(b.deletes, string(key))
+	b.size += len(key)
+	return nil
+}
+
+// ValueSize retrieves the amount of data queued up for writing.
+func (b *batch) ValueSize() int {
+	return b.size
+}
+
+// Write flushes any accumulated data to the memory database.
+func (b *batch) Write() error {
+	if len(b.deletes) > 0 {
+		err := b.db.client.Del(b.deletes...).Err()
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(b.writes) > 0 {
+		err := b.db.client.MSet(b.writes).Err()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
This method does not respect the sequential semantics of a batch. When using a batch, performing `n` sequential operations, it's expected that the order is maintained. You are doing all the deletes first, and then the writes. 

If we don't have it already, then we definitely should add a testcase for this in the db test suite.

Reply:
Good point, @holiman, thank you. I didn't understand that the sequence was so essential and I was trying to improve performance. 
One doubt: if I guarantee the sequence order, can I aggregate similar operations (delete/set) and send them together? Avoiding multiple single remote requests...

Reply:
It is definitely allowable to aggregate the results locally, and only send the 'outcome' as a batch.

Reply:
> if I guarantee the sequence order, can I aggregate similar operations (delete/set) and send them together? A

Ah, sorry, misunderstood the context. Then I agree with what @fjl said

Reply:
@holiman , fixed. May you review it again? Thanks